geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/factory/FastFilterFactory.scala

```diff
@@ -224,7 +224,7 @@ class FastFilterFactory private extends org.geotools.filter.FilterFactoryImpl wi
                            exp2: Expression,
                            matchCase: Boolean,
                            matchAction: MatchAction): PropertyIsLessThanOrEqualTo = {
-    if (matchCase || matchAction != MatchAction.ANY) {
+    if (matchAction != MatchAction.ANY) {
```

Comment:
looking at the geotools filter impls, they seem to ignore `matchCase`, so this seems safe enough.